llvm/lib/IR/Instructions.cpp

```diff
@@ -5196,7 +5198,11 @@ void SwitchInstProfUpdateWrapper::init() {
   if (!ProfileData)
     return;
 
-  if (ProfileData->getNumOperands() != SI.getNumSuccessors() + 1) {
+  // FIXME: This check belongs in ProfDataUtils. Its almost equivalent to
+  // getValidBranchWeightMDNode(), but the need to use llvm_unreachable
+  // makes them slightly different.
+  if (ProfileData->getNumOperands() !=
+      SI.getNumSuccessors() + getBranchWeightOffset(ProfileData)) {
```

Comment:
This seems simple enough to do something about it? Could for example add a `getNumBranchWeights(<profile_data>)` API so this can become `getNumBranchWeights(ProfileData) != SI.getNumSuccessors()`?

Reply:
Another good suggestion. Thank you.